app/Http/Controllers/API/User/PasswordChangeController.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brewmap\Http\Controllers\API\User;
+
+use Brewmap\Exceptions\User\OldPasswordMismatchException;
+use Brewmap\Http\Controllers\Controller;
+use Brewmap\Http\Requests\User\UpdatePasswordRequest;
+use Brewmap\Services\UpdatePasswordService;
+use Illuminate\Http\JsonResponse;
+
+class PasswordChangeController extends Controller
+{
+    /**
+     * Update the user password with cofirmation old password.
+     *
+     * @throws OldPasswordMismatchException
+     */
+    public function update(UpdatePasswordRequest $request, UpdatePasswordService $service): JsonResponse
+    {
+        $service->updatePassword($request->validated());
+        return response()->json([
```

Comment:
You can try to use `lluminate\Contracts\Routing\ResponseFactory` or just simply return new `JsonResponse` instance instead of using `response()` helper.

```suggestion
        return $this->responseFactory->json([
```

```suggestion
        return new JsonResponse([
```

Reply:
It will be change in all controllers for API?

Reply:
Of course. We are doing everything one way.